### PR TITLE
fix: limit to the first 20 enrs for peering only

### PR DESCRIPTION
**Pull request by barnabasbusa**

### CODE REVIEW

File: src/cl/lighthouse/lighthouse_launcher.star [hunk 1]
```diff
@@ -247,8 +247,8 @@ def get_beacon_config(
 	]
 
 	if boot_cl_client_ctxs != None:
-		cmd.append("--boot-nodes="+",".join([ctx.enr for ctx in boot_cl_client_ctxs]))
-		cmd.append("--trusted-peers="+",".join([ctx.peer_id for ctx in boot_cl_client_ctxs]))
+		cmd.append("--boot-nodes="+",".join([ctx.enr for ctx[:20] in boot_cl_client_ctxs]))
```
Comment (h4ck3rk3y) [inline src/cl/lighthouse/lighthouse_launcher.star:250]: can we take this out in a constant that we share across the nodes? Easier to update & think of that way.

File: src/cl/nimbus/nimbus_launcher.star [hunk 2]
```diff
@@ -240,7 +240,7 @@ def get_config(
 		# See explanation there
 		cmd.append("--subscribe-all-subnets")
 	else:
-		for ctx in bootnode_contexts:
+		for ctx[:20] in bootnode_contexts:
```
Comment (h4ck3rk3y) [inline src/cl/nimbus/nimbus_launcher.star:243]: this should read

`for ctx in bootnode_contexts[:20]`

File: src/cl/lodestar/lodestar_launcher.star [hunk 3]
```diff
@@ -230,7 +230,7 @@ def get_beacon_config(
 	]
 
 	if bootnode_contexts != None :
-		cmd.append("--bootnodes="+",".join([ctx.enr for ctx in bootnode_contexts]))
+		cmd.append("--bootnodes="+",".join([ctx.enr for ctx[:20] in bootnode_contexts]))
```
Comment (h4ck3rk3y) [inline src/cl/lodestar/lodestar_launcher.star:233]: this should read bootnode_contexts[:20] instead

File: src/cl/teku/teku_launcher.star [hunk 4]
```diff
@@ -230,8 +230,8 @@ def get_config(
 	]
 
 	if bootnode_contexts != None:
-		cmd.append("--p2p-discovery-bootnodes="+",".join([ctx.enr for ctx in bootnode_contexts]))
-		cmd.append("--p2p-static-peers="+",".join([ctx.multiaddr for ctx in bootnode_contexts]))
+		cmd.append("--p2p-discovery-bootnodes="+",".join([ctx.enr for ctx[:20] in bootnode_contexts]))
+		cmd.append("--p2p-static-peers="+",".join([ctx.multiaddr for ctx[:20] in bootnode_contexts]))
```
Comment (h4ck3rk3y) [inline src/cl/teku/teku_launcher.star:234]: same as above

File: src/cl/prysm/prysm_launcher.star [hunk 5]
```diff
@@ -253,7 +253,7 @@ def get_beacon_config(
 	]
 
 	if bootnode_contexts != None:
-		for ctx in bootnode_contexts:
+		for ctx[:20] in bootnode_contexts:
```
Comment (h4ck3rk3y) [inline src/cl/prysm/prysm_launcher.star:256]: same as above

**Review by h4ck3rk3y — CHANGES_REQUESTED** [review]

File: src/cl/lighthouse/lighthouse_launcher.star (hunk 1 above)
Comment (barnabasbusa) [reply src/cl/lighthouse/lighthouse_launcher.star]: I wouldn't worry about this tbh.

Comment (h4ck3rk3y) [reply src/cl/lighthouse/lighthouse_launcher.star]: i worry about it as I don't want magical numbers floating around; next time we change it we have to change 20 everywhere!

**Review by h4ck3rk3y — APPROVED** [review]

File: src/cl/lighthouse/lighthouse_launcher.star (hunk 1 above)
Comment (barnabasbusa) [reply src/cl/lighthouse/lighthouse_launcher.star]: fixed.

**Pull request merged**